filters/openpolicyagent/internal/envoy/skipperadapter.go

```diff
@@ -0,0 +1,34 @@
+package envoy
+
+import (
+	"net/http"
+	"strings"
+
+	ext_authz_v3_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	ext_authz_v3 "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+)
+
+func AdaptToEnvoyExtAuthRequest(req *http.Request, metadata *ext_authz_v3_core.Metadata, contextExtensions map[string]string) *ext_authz_v3.CheckRequest {
+
+	headers := make(map[string]string)
+	for h, vv := range req.Header {
+		// This makes headers in the input compatible with what Envoy does, i.e. allows to use policy fragments designed for envoy
+		// See: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/header_casing#http-1-1-header-casing
+		headers[strings.ToLower(h)] = strings.Join(vv, ", ")
```

Comment:
Is it mandatory to copy all headers for something to work or is this nice to have?
For me this creates a lot of data by maybe even untrusted clients that could use it to create GC pressure.

I guess we need it because, but feels bad:
```
% go doc github.com/envoyproxy/go-control-plane/envoy/service/auth/v3.AttributeContext_HttpRequest.Headers
package authv3 // import "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"

type AttributeContext_HttpRequest struct {
    // The HTTP request headers. If multiple headers share the same key, they must
    // be merged according to the HTTP spec. All header keys must be lower-cased,
    // because HTTP header keys are case-insensitive.
    Headers map[string]string

    // ... other fields elided ...
}
```

Reply:
To support the http spec fully and support repeated headers in policies, I fear we need it.